core/framework/metrics.py

```
@@ -0,0 +1,182 @@
+        Record an error.
+        
+        Args:
+            error: Error message
+        """
+        with self.lock:
+            self.metrics['errors'].append({
+                'timestamp': datetime.now().isoformat(),
+                'error': str(error)
+            })
+    
+    def get_summary(self) -> Dict[str, Any]:
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)